proof/mint.go

```diff
@@ -11,9 +11,86 @@ import (
 	"github.com/lightninglabs/taproot-assets/commitment"
 )
 
-// Blob represents a serialized proof file, including the checksum.
+// Blob either represents a serialized proof file, including the checksum or a
+// single serialized issuance/transition proof. Which one it is can be found out
+// from the leading magic bytes (or the helper methods that inspect those).
```

Comment:
Hmm, I wonder if we would be better of making these into different types instead. We added the magic prefix bytes partially to avoid confusing them in the first place. With the type route, we'd break compilation if one was attempted to be passed in where the other was accepted. As is, we reply on runtime checks to avoid such classes of bugs.

Reply:
This could use the `Either` type added in the multiverse root PR as an example. Then you can still use `Blob` everywhere, but rely on type safety re protection of trying to use one vs the other.

Reply:
Yes, I think distinguishing between a single proof or a proof file would help a lot in the clarity of the code. But I think that would be a bigger refactor and should probably be done in a separate PR.
Do you think this is a reasonable temporary step for this PR?

Reply:
SGTM, down to make an issue for it so we can move on here. Main thing is we can supplement the byte level prefix with a type, and then accept that type contextually to cut down on possible bugs in this area.